### ISSUE TITLE
WithClauseError   Ueberauth.Strategy.Apple.handle_callback!/1

### ISSUE THREAD

**Issue by alvnrapada** [ueberauth/ueberauth_apple#2]
### Steps to Reproduce

I'm trying to implement sign in with apple on our app - which I did and tested, its working on 3 of my apple accounts but when I ask my coworkers to test it, it doesn't work. 

### Expected Result

 ```
 with {:ok, %{"email" => email, "sub" => uid}} <- Token.payload(token, token_opts),
```

its expecting a result with email

### Actual Result

(WithClauseError no with clause matching: {:ok, %{"aud" => "asdfasdfasdfasdfasdf", "auth_time" => 1686319595, "c_hash" => "asdfasdfasdfasdfasdfasdf", "exp" => 1686405995, "iat" => 1686319595, "iss" => "https://appleid.apple.com", "nonce_supported" => true, "sub" => "asdfasdfasdfasdfasdfasdfasdfasdf"}})

But it does not have one.

**Comment by aj-foster** [comment]
Hi @alvnrapada, thanks for reporting this. I have a proposed fix on the `aj/fix-missing-email-error` branch. If you have a chance, would you be able to test it out and see if it fixes the issue?

**Comment by aj-foster** [comment]
Also, since you have a few test cases: is it possible that it works for people who choose to hide their email (and therefore have a private relay address), and it doesn't work for people who share their real email address? Just a theory of mine.

**Comment by alvnrapada** [comment]
> Also, since you have a few test cases: is it possible that it works for people who choose to hide their email (and therefore have a private relay address), and it doesn't work for people who share their real email address? Just a theory of mine.

I don't think so - I have tried both on 3 of my apple accounts in two laptops and all were working. But it never worked on my co-workers.

**Comment by alvnrapada** [comment]
> Hi @alvnrapada, thanks for reporting this. I have a proposed fix on the `aj/fix-missing-email-error` branch. If you have a chance, would you be able to test it out and see if it fixes the issue?

Unfortunately, I moved out of `ueberauth_apple`,  and made it work following their Sign in with Apple JS documentation.

**Comment by aj-foster** [comment]
Resolved in release 0.6.1. ❤️
